plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerManager.kt

```diff
@@ -477,7 +479,15 @@ class CodeModernizerManager(private val project: Project) : PersistentStateCompo
             }
 
             LOG.info { "Attempting to resume job, current state is: $managerState" }
-            if (!managerState.flags.getOrDefault(StateFlags.IS_ONGOING, false)) return@launch
+            if (!managerState.isJobOngoing()) return@launch
+
+            val lastJobId = managerState.getLatestJobId()
+            val isJobExpired = managerState.getLatestJobStartTime().plusMillis(JOB_EXPIRY_TIME_DAYS.days.inWholeMilliseconds) < Instant.now()
+            if (isJobExpired) {
+                LOG.info { "Last job is expired, won't attempt to resume job with Id $lastJobId anymore" }
```

Comment:
Blocker/Question:

Just need to finalize the message string and if this message needs to:
1. Be displayed to users
2. Logged to .idea logs
3. Other

Reply:
1. Currently this is a silent operation and we only inform customers on success, so don't think we should notify users here
2. Probably we don't have to log this, i'll remove this
3. Not sure what Other means :D